### PR TITLE
First pass at adding frame chain to NaifSpice class

### PR DESCRIPTION
**Pull request by SgStapleton**
I have not figured out yet how to access the spacecraft frame NAIF ID yet, but once I get that figured out, I will get it implemented.

I wanted to get this up for feedback as soon as possible, though. I went with a dictionary format since any NaifSpice object was going to have the same four frame nodes (j2000, spacecraft, sensor, and target).

Any feedback appreciated! This references https://github.com/USGS-Astrogeology/ale/issues/164

### CODE REVIEW

File: ale/base/data_naif.py [hunk 1]
```diff
@@ -150,10 +151,22 @@ def sensor_position(self):
 
     @property
     def frame_chain(self):
-        pass
-
-    @property
-    def _sensor_orientation(self):
+        # pass
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Remove old commented pass statement

File: ale/base/data_naif.py [hunk 2]
```diff
@@ -150,10 +151,22 @@ def sensor_position(self):
...
+            frame_chain['j2000'] = FrameNode(1)
+            # self._frame_chain['spacecraft'] = FrameNode(self.)
+            frame_chain['sensor'] = FrameNode(self.sensor_frame_id, 
+                                                parent=frame_chain['j2000'],
+                                                rotation='ConstantRotation')
+            frame_chain['target'] = FrameNode(self.target_frame_id, 
+                                                 parent=frame_chain['j2000'],
+                                                 rotation='TimeDependentRotation')
+            self._frame_chain = frame_chain
+        return self._frame_chain
```
Comment (jessemapel) [inline ale/base/data_naif.py]: The Frame Chain is a tree object, so we can just return the root node of the tree (99.99999% of the time with SPICE this is the J2000 frame).

File: ale/base/data_naif.py [hunk 3]
```diff
@@ -150,10 +151,22 @@ def sensor_position(self):
...
-    @property
-    def _sensor_orientation(self):
+        # pass
+        if not hasattr(self, '_frame_chain'):
+            frame_chain = {}
+            frame_chain['j2000'] = FrameNode(1)
+            # self._frame_chain['spacecraft'] = FrameNode(self.)
+            frame_chain['sensor'] = FrameNode(self.sensor_frame_id, 
+                                                parent=frame_chain['j2000'],
+                                                rotation='ConstantRotation')
```
Comment (jessemapel) [inline ale/base/data_naif.py]: This should be an actual ConstantRotation or TimeDependentRotation object, see rotation.py. You can use the [pxform](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/pxform.html) function to get the rotation matrix.

File: ale/base/data_naif.py [hunk 4]
```diff
@@ -150,10 +151,22 @@ def sensor_position(self):
...
+            frame_chain['target'] = FrameNode(self.target_frame_id, 
+                                                 parent=frame_chain['j2000'],
+                                                 rotation='TimeDependentRotation')
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Same as above, this should be one of the two Rotation objects defined in rotations.py

File: ale/base/data_naif.py (hunk 3 above)
Comment (SgStapleton) [reply ale/base/data_naif.py]: Just to make sure I am fully understanding this: this rotation matrix should be able to be passed straight into the construction of a rotation object as the "quat" value, right?

File: ale/base/data_naif.py [hunk 5]
```diff
@@ -150,10 +151,22 @@ def sensor_position(self):
...
     @property
     def frame_chain(self):
-        pass
-
-    @property
-    def _sensor_orientation(self):
+        # pass
+        if not hasattr(self, '_frame_chain'):
+            frame_chain = {}
+            frame_chain['j2000'] = FrameNode(1)
```
Comment (SgStapleton) [inline ale/base/data_naif.py]: We should be able to safely assume in this implementation that the root of this tree will always be J2000, right?

**Comment by jessemapel** [comment]
Continued at #181

**Pull request closed without merging**